File: CS-Examples/20_Print/PrintExcel.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;
using System.Drawing.Printing;

namespace PrintExcel
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            //Load the Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\PrintExcel.xlsx");

            // Access the printer settings of the workbook's print document
            PrinterSettings settings = workbook.PrintDocument.PrinterSettings;

            // Specify the range of pages to be printed (from page 0 to page 1)
            settings.FromPage = 0;
            settings.ToPage = 1;

            // Use the default printer to print
            workbook.PrintDocument.Print();

            // Dispose of the workbook object to release resources
            workbook.Dispose();
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/20_Print/PrintExcel.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports System.Drawing.Printing

Namespace PrintExcel
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\PrintExcel.xlsx")

            ' Get the PrinterSettings object from the workbook's PrintDocument
            Dim settings As PrinterSettings = workbook.PrintDocument.PrinterSettings

            ' Set the starting page number for printing to 0 (first page)
            settings.FromPage = 0

            ' Set the ending page number for printing to 1 (second page)
            settings.ToPage = 1

            ' Print the workbook's contents using the configured printer settings
            workbook.PrintDocument.Print()

            ' Release the resources used by the workbook
            workbook.Dispose()
        End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
